gcloud/datastore/query.py

```diff
@@ -463,6 +463,11 @@ def __iter__(self):
                 yield entity
             if not self._more_results:
                 break
+            num_results = len(self._page)
+            if self._limit is not None:
+                self._limit -= num_results
+            if self._offset is not None:
+                self._offset += num_results
```

Comment:
Correct. `offset` should **never** be used to continue a query as it is expensive and has a limit to the amount of offset a backend will satisfy in a single request.